internal/pkg/cli/storage_init.go

```diff
@@ -692,65 +691,63 @@ func (o *initStorageOpts) readWorkloadType() error {
 }
 
 type addonBlob struct {
-	name        string
 	description string
 	blob        encoding.BinaryMarshaler
+	path        string
 }
 
 func (o *initStorageOpts) addonBlobs() ([]addonBlob, error) {
-	templateBlob, err := o.newAddonTemplate()
+	switch {
+	case o.lifecycle == lifecycleWorkloadLevel && o.storageType == s3StorageType:
```

Comment:
nit: I was experimenting with this model of writing so that there is no conditional, what do you think? 
https://go.dev/play/p/hMJ3YcffM0x

It's really a nit

Reply:
The `switch selection` kind of writing seems to end up being as efficient as the other, if not better 🤔 It is more readable as well. I will update the code.

Reply:
done!